### PR TITLE
Smart Wallets: Export Types From `react-ui` Package + Cleanup

### PR DESCRIPTION
**Pull request by devlyn37**
## Description

This PR:
- Removes uneeded deps from the new smart wallet demo
- Exports a couple smart wallet types from `@crossmint/client-sdk-react-ui` that we're needed as a result.

## Test plan

Went through the demo flow, logged in, minted an NFT.

### CODE REVIEW

File: apps/wallets/smart-wallet/next/src/env.ts
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/env.ts:1]: This is overkill for our single env var

File: packages/client/ui/react-ui/src/index.ts [hunk 1]
```diff
@@ -3,4 +3,5 @@ export * from "./hooks";
 export * from "./providers";
 
 export { CrossmintEvents, useCrossmintEvents } from "@crossmint/client-sdk-base";
+export type { EVMSmartWallet, Chain } from "@crossmint/client-sdk-smart-wallet";
```
Comment (devlyn37) [inline packages/client/ui/react-ui/src/index.ts:6]: Removing `@crossmint/client-sdk-smart-wallet` as a dependency from the demo package revealed we need to export these.

**Review by jmderby — APPROVED** [review]
Nice, cleaned up quite a bit

File: apps/wallets/smart-wallet/next/src/env.ts
Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/env.ts:1]: good call

**Pull request merged**